src/state/validation.py

```diff
@@ -60,13 +61,15 @@ def wrapper(instance: C, *args: typing.Any) -> None:
             except (
                 state.config.InvalidCharmConfigError,
                 state.tls.TlsIntegrationMissingError,
+                state.config.GatewayClassUnavailableError,
+                InsufficientPermissionError,
             ) as exc:
                 if defer:
                     event: ops.EventBase
                     event, *_ = args
                     event.defer()
-                logger.exception("Wrong Charm Configuration")
-                instance.unit.status = ops.BlockedStatus(exc.msg)
+                logger.exception(str(exc))
```

Comment:
```suggestion
                logger.exception("Invalid user configuration exception.")
```
I think since logger.exception already prints the exception stack trace, re-printing it would be a bit too verbose :D

Reply:
thanks, will add this in a follow-up PR